backend/src/main/java/coffeeshout/room/infra/messaging/handler/RoomCreateEventHandler.java

```diff
@@ -16,30 +18,90 @@ public class RoomCreateEventHandler implements RoomEventHandler<RoomCreateEvent>
 
     private final RoomService roomService;
     private final RoomEventWaitManager roomEventWaitManager;
+    private final RedisTemplate<String, String> redisTemplate;
 
     @Override
     public void handle(RoomCreateEvent event) {
         try {
             log.info("방 생성 이벤트 수신: eventId={}, hostName={}, joinCode={}",
                     event.eventId(), event.hostName(), event.joinCode());
 
-            final Room room = roomService.createRoomInternal(
-                    event.hostName(),
-                    event.selectedMenuRequest(),
-                    event.joinCode()
-            );
+            final Room room = createRoomInMemory(event);
 
-            roomEventWaitManager.notifySuccess(event.eventId(), room);
+            tryDbSave(event, room);
 
-            log.info("방 생성 이벤트 처리 완료: eventId={}, joinCode={}",
-                    event.eventId(), event.joinCode());
+            roomEventWaitManager.notifySuccess(event.eventId(), room);
 
         } catch (Exception e) {
             log.error("방 생성 이벤트 처리 실패", e);
             roomEventWaitManager.notifyFailure(event.eventId(), e);
         }
     }
 
+    private Room createRoomInMemory(RoomCreateEvent event) {
+        return roomService.createRoomInternal(
+                event.hostName(),
+                event.selectedMenuRequest(),
+                event.joinCode()
+        );
+    }
+
+    private void tryDbSave(RoomCreateEvent event, Room room) {
+        final String lockKey = "event:lock:" + event.eventId();
+        final String doneKey = "event:done:" + event.eventId();
+
+        if (isAlreadyProcessed(doneKey, event.eventId())) {
+            return;
+        }
+
+        if (!acquireLock(lockKey, event.eventId())) {
+            return;
+        }
```

Comment:
요렇게 하면 더 간결해질 것 같습니다!
```suggestion
        if (isAlreadyProcessed(doneKey, event.eventId()) || !acquireLock(lockKey, event.eventId())) {
            return;
        }
```